### PR TITLE
[SPARK-39355][SQL] Single column uses quoted to construct UnresolvedAttribute

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
Use `UnresolvedAttribute.quoted` in `Alias.toAttribute` to avoid calling `UnresolvedAttribute.apply` causing `ParseException`.

### Why are the changes needed?
```sql
SELECT *
FROM (
    SELECT '2022-06-01' AS c1
) a
WHERE c1 IN (
    SELECT date_add('2022-06-01', 0)
);
```
```
Error in query:
mismatched input '(' expecting {<EOF>, '.', '-'}(line 1, pos 8)
== SQL ==
date_add(2022-06-01, 0)
--------^^^ 
```

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
add UT

### CODE REVIEW

**Comment by cxzl25** [comment]
@sarutak @cloud-fan @dongjoon-hyun

After the introduction of SPARK-34636, some SQL will fail to parse.

**Comment by cxzl25** [comment]
Gentle ping @sarutak @cloud-fan @dongjoon-hyun
This should be a bug, hope to help review.

File: sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala [hunk 1]
```diff
@@ -2176,4 +2176,32 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-39355: Avoid UnresolvedAttribute.apply throwing ParseException") {
+    checkAnswer(
+      sql("""
+            |SELECT *
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala:2183]: I'm surprised this bug causes real query failures. Can you explain how this query triggers `UnresolvedAttribute.apply`?

Comment (cxzl25) [reply sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala:2183]: A `Cast` may be generated `InConversion`, at which time `newSub.output` will call `UnresolvedAttribute.apply`.
https://github.com/apache/spark/blob/d3d22928d4fca1cd71f96a1308f0cc5d00120ad5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala#L349-L360

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala:2183]: I think we should fix `Alias.toAttribute`. It should create an attribute to reference the output of this alias, which is a single column, and we need to use `UnresolvedAttribute.quoted`.

Comment (cxzl25) [reply sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala:2183]: It seems that there may be a problem here, although I didn't reproduce it using SQL, do I need to fix it?

https://github.com/apache/spark/blob/2349175e1b81b0a61e1ed90c2d051c01cf78de9b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala#L120

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala:2183]: yea we should also use `.quoted` there. good catch!

**Comment by cloud-fan** [comment]
thanks, merging to master/3.3/3.2!

**Pull request closed without merging**